read-fonts/src/tables/variations.rs

```diff
@@ -11,6 +11,479 @@ pub struct DeltaSetIndex {
     pub inner: u16,
 }
 
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct TupleIndex(u16);
+
+impl TupleIndex {
+    /// Flag indicating that this tuple variation header includes an embedded
+    /// peak tuple record, immediately after the tupleIndex field.
+    ///
+    /// If set, the low 12 bits of the tupleIndex value are ignored.
+    ///
+    /// Note that this must always be set within the 'cvar' table.
+    const EMBEDDED_PEAK_TUPLE: u16 = 0x8000;
+
+    /// Flag indicating that this tuple variation table applies to an
+    /// intermediate region within the variation space.
+    ///
+    /// If set, the header includes the two intermediate-region, start and end
+    /// tuple records, immediately after the peak tuple record (if present).
+    const INTERMEDIATE_REGION: u16 = 0x4000;
+    /// Flag indicating that the serialized data for this tuple variation table
+    /// includes packed “point” number data.
+    ///
+    /// If set, this tuple variation table uses that number data; if clear,
+    /// this tuple variation table uses shared number data found at the start
+    /// of the serialized data for this glyph variation data or 'cvar' table.
+    const PRIVATE_POINT_NUMBERS: u16 = 0x2000;
+    //0x1000	Reserved	Reserved for future use — set to 0.
+    //
+    /// Mask for the low 12 bits to give the shared tuple records index.
+    const TUPLE_INDEX_MASK: u16 = 0x0FFF;
+
+    fn tuple_len(self, axis_count: u16, flag: usize) -> usize {
+        match flag {
+            0 => self.embedded_peak_tuple(),
+            1 => self.intermediate_region(),
+            _ => panic!("only 0 or 1 allowed here"),
```

Comment:
I'm not sure if the flag comes from the font or user. If from the font, then panic is not desired. If from the user, maybe an enum is better? @rsheeter

Reply:
This flag comes from the codegen input, and the codegen tries to use a language-agnostic schema. In this particular corner of codegen, (when calculating a count) the only inputs that can be provided are unsigned integers, so I'm overloading that in this particular instance to represent an enum. 😒